File: src/FoodPal.Providers.DomainModels/Provider.cs
using System.ComponentModel.DataAnnotations;

namespace FoodPal.Providers.DomainModels
{
    public class Provider : BaseEntity
    {
        [Required]
        [MaxLength(150)]
        public string Name { get; set; }
        [Required]
        [MaxLength(500)]
        public string Description { get; set; }
        [Required]
        [MaxLength(300)]
        public string Location { get; set; }
        [Required]
        public Catalogue Catalogue { get; set; }

        public ProviderCategory Category { get; set; }

        public int CategoryId { get; set; }
        [Required]
        public int CustomerId { get; set; }

    }
}

File: food-pal/src/app/admin/models/provider.ts
import { ProviderCatalogue } from './provider-catalogue';
import { ProviderCategory } from './provider-category';

export class Provider {
  id: number;
  name: string;
  description: string;
  location: string;
  customerId: number = 1;
  catalogue: ProviderCatalogue;
  category?: ProviderCategory;
}